src/invoices/InvoiceDetails/ApproveConfirmationModal/useDuplicateInvoice.js

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+import moment from 'moment';
+import { keyBy } from 'lodash';
+
+import {
+  batchFetch,
+  LIMIT_MAX,
+  DATE_FORMAT,
+  useShowCallout,
+} from '@folio/stripes-acq-components';
+
+const getInvoices = async (mutator, invoice) => {
+  const formattedInvoiceDate = moment.utc(invoice.invoiceDate)?.format(DATE_FORMAT);
+
+  try {
+    return await mutator.duplicateInvoices.GET({
+      limit: `${LIMIT_MAX}`,
+      params: {
+        query: `id<>"${invoice.id}" AND
+          vendorInvoiceNo=="${invoice.vendorInvoiceNo}" AND
+          invoiceDate=="${formattedInvoiceDate}*" AND
+          vendorId=="${invoice.vendorId}"`,
+      },
+    });
+  } catch {
+    throw new Error('ui-invoice.invoice.actions.load.error');
+  }
+};
+
+const getVendors = (mutator, vendorIds) => batchFetch(mutator.vendors, vendorIds).catch(() => []);
+
+const getAll = async (mutator, invoice) => {
+  const invoices = await getInvoices(mutator, invoice);
+  const vendorIds = invoices.map(inv => inv.vendorId);
+  const uniqueVendorIds = [...new Set(vendorIds)];
+  const vendors = await getVendors(mutator, uniqueVendorIds);
+
+  const vendorMap = keyBy(vendors, 'id');
+
+  return invoices.map(inv => ({
+    ...inv,
+    vendor: vendorMap[inv.vendorId],
+  }));
+};
+
+const useDuplicateInvoice = (mutator, invoice) => {
+  const showCallout = useShowCallout();
+  const [invoices, setInvoices] = useState();
+
+  useEffect(
+    () => {
+      getAll(mutator, invoice)
+        .then(setInvoices)
+        .catch(({ message }) => showCallout({ messageId: message, type: 'error' }));
```

Comment:
`useDuplicateInvoice` is now used in modal, so probably no problems with previous state.
But what if it's used anywhere else, where it could get new `invoice`, and keep the state?
The API error would lead to inconsistency: new `invoice`,  old state.
So maybe `setInvoices(undefined)` ?

Reply:
just `setInvoices()` should be enougth

Reply:
great!
explicit param is just more readable for my taste

Reply:
On second thought, it's a double state change in success case, so double render.
Not much, but...